Administrator Guide/Split brain and ways to deal with it.md

```diff
@@ -0,0 +1,61 @@
+
+### Replica 2 volume:
+In a replica 2 volume it is not possible to achieve high availability and consistency at the same time, without sacrificing tolerance to partition. If we set the client-quorum option to auto, then the first brick must always be up, irrespective of the status of the second brick. If only the second brick is up, the subvolume becomes EROFS. If the quorum-type is set to fixed, then the quorum-count is 1, then we may end up in split brain. i.e., One write succeed on one brick and other write on another brick. To avoid this we have to set the quorum-count to 2, which will cost the availability. Even if we have one replica brick up and running, we will end up seeing EROFS.
+
+#### 1. Replica 3 volume:
+When we create a replicated or distributed replicated volume with replica count 3, the cluster.quorum-type option is set to auto by default. That means at least 2 bricks should be up and running to satisfy the quorum and allow the writes. This is the recommended setting for a replica 3 volume and this should not be changed. Here is how it prevent files from ending up in split brain:
+
+B1, B2, and B3 are the 3 bricks of a replica 3 volume.
+1. B1 & B2 are up and B3 is down. Quorum is met and write happens on B1 & B2. Pending xattrs for B3 is set on B1 & B2.
+2. B3 comes up and B2 is down. Quorum is met and write happens on B1 & B3 and pending xattrs are set for B2 on B1 & B3.
+3. B2 comes up and B1 goes down. Quorum is met. But when a write request comes, AFR sees that B2 & B3's pending xattrs blame each other and therefore the write is not allowed and is failed with EIO.
+
+Command to create a replica 3 volume:
+```sh
+$gluster volume create <volname> replica 3 host1:brick1 host2:brick2 host3:brick3
+```
+There is a corner case even with replica 3 volumes where the file can end up in a split-brain. AFR usually takes range locks for the {offset, length} of the write. If 3 writes happen on the same file at non-overlapping {offset, length} and each write fails on (only) one different brick, then we have AFR xattrs of the file blaming each other.
+
+### 2. Arbiter volume:
+Arbiter offers the sweet spot between replica 2 and replica 3, where user wants the split-brain protection offered by replica 3 but does not want to invest in 3x storage space. Arbiter is also an replica 3 volume where the third brick of the replica is automatically configured as an arbiter node. This means that the third brick stores only the file name and metadata, but not any data. This will help in avoiding split brain while providing the same level of consistency as a normal replica 3 volume.

```

Comment:
That document is more tilted towards arbiter volumes. This document explains arbiter and replica 3 in equal detail. In this, I have provided link to that document for more details on arbiter.
Link to this document will be given in the warning message to the user, wile creating a replica 2 volume.